README.md

```diff
@@ -1 +1,139 @@
+[OpenAPI 3.0 Specification](https://github.com/OAI/OpenAPI-Specification/blob/3.0.0-rc0/versions/3.0.md),
+and a highly uniform read/write programming API.
+
+Feature highlights include:
+* **High Performance** - Informal testing shows a 3x-4x performance
+  improvement over the current Java Swagger 2.0 parser. This is
+  largely attributable to a design based on adapting Jackson
+  `JsonNode` objects, rather than deserializing to internal POJOs.
+  
+* **Read/Write API** - All aspects of a model may be interrogated and
+  modified through the API. We also provide fluent builders for all
+  API object types. API provides bidirectional navigation through the
+  model, and every object that is a property value of its containing
+  object (whether as a named field or a map entry) knows its own key.
+  
+* **Tolerant Reader** - The parser yields a fully accessible result
+  from any valid JSON or YAML input - whether or not the input is a
+  valid OpenAPI specification.
+  
+* **Separate, extensible validation** - All validation beyond base
+  JSON/YAML parsing is performed after the initial parse, and it can
+  be disabled for speed. API-level validation is separated from
+  injectable implementation-level validation. The latter, for example,
+  is where unrecognized JSON properties can be noted, as these are not
+  visible from the model API.
+
+* **Serialization** - Serialization to JSON or YAML is supported, and
+  by default, round-tripping will not cause any reordering of model
+  content.
+  
+* **DI-Ready** - Our API is interface-based and allows substitution of
+  custom implementations that either augment or completely replace our
+  provided implementation, using
+  [Google Guice](https://github.com/google/guice) dependency
+  injection.
+  
+* **Easy Evolution** - A YAML-based DSL is used to capture most of the
+  details of the OpenAPI Specification. We use code generation to
+  create interfaces and implementation classes. Member-level
+  `@Generated` annotations make it possible to augment the generated
+  sources with manual code that is preserved during re-generation.
+* **Flexible Reference Handling** - References are detected and
+  resolved after JSON/YAML parsing but before model-level parsing. All
+  JSON references are processed, resulting in effective inlining for
+  non-conforming references (i.e. anything other than path or
+  component references specifically allowed in the
+  specification). References are normally traversed automatically by
+  the API, but full details of references and their resolution status
+  are also available.
+  
+## Current State
+
+* `JsonOverlay` framework is fairly stable and working well.
+
+* OpenAPI 3.0 type specification, and the generated code passes some
+  early tests.
+
+* Both read and write APIs are fully implemented, though not well
+  tested. No builders have been created as yet.
+
+* Model-level validation is mostly complete, but requires development
```

Comment:
Yes. "API-Level" means that everything detected here is visible through the Java API. I've used the term "model-level" in some cases more or less synonymously. If "model-level" is clearer I'll switch to that. Implementation-level means everything else, like the example I gave above - noting unrecognized properties appearing in the spec. (And doing whatever the parser is configured to do with those - error, warning or ignore, which is why I said "noted" rather than being more specific.)

The API-level validation is implemented universally, while implementation-level validation, to the extent it's offered, needs to be injected, and would go outside the programming API to do things the API can't do - like view the overlaid JsonNode objects in the case of our implementation.